### DISCUSSION TITLE
How to manually register users with built-in auth?

### DISCUSSION THREAD

**Discussion by notramo** [geldata/gel#6513] [Q&A]
I'm using the built-in auth with the default UI. I want to disable registration, and manually register a few email-password users. Is it possible?

**Comment by raddevon** [comment]
Had to revise the previous method as we realized why it won't work. 😅 

Instead, your best bet is to leave the provider on but not expose the UI. Then, you can make a request to the registration endpoint for your new user with a randomly generated password. Make a request to the send-reset-email endpoint to send the password reset email to your new user. They can then set their password.

In order to make this work though, you won't be able to use any of the built-in UI. You'll have to create your own UI for logging in. You won't be able to expose the EdgeDB server either since that will leave your registration endpoint open.

**Reply by notramo** [reply]
I want to use the built-in UI, but not for registrations. Logins would still go through it.

**Reply by raddevon** [reply]
It's not currently supported. You may want to file a feature request.
